src/app/AttributeCache.cpp

```diff
@@ -56,25 +94,33 @@ CHIP_ERROR AttributeCache::UpdateCache(const ConcreteDataAttributePath & aPath,
     // if the endpoint didn't exist previously, let's track the insertion
     // so that we can inform our callback of a new endpoint being added appropriately.
     //
-    if (mCache.find(aPath.mEndpointId) == mCache.end())
+    if (findNewEndpoint)
     {
         mAddedEndpoints.push_back(aPath.mEndpointId);
     }
 
-    mCache[aPath.mEndpointId][aPath.mClusterId][aPath.mAttributeId] = std::move(state);
+    mCache[aPath.mEndpointId][aPath.mClusterId].mAttributes[aPath.mAttributeId] = std::move(state);
     mChangedAttributeSet.insert(aPath);
     return CHIP_NO_ERROR;
 }
 
 void AttributeCache::OnReportBegin()
 {
+    mLastConcreteClusterPath = ConcreteClusterPath(kInvalidEndpointId, kInvalidClusterId);
     mChangedAttributeSet.clear();
     mAddedEndpoints.clear();
     mCallback.OnReportBegin();
 }
 
 void AttributeCache::OnReportEnd()
 {
+    if (mLastConcreteClusterPath.IsValidConcreteClusterPath())
+    {
+        mCache[mLastConcreteClusterPath.mEndpointId][mLastConcreteClusterPath.mClusterId].mCommittedDataVersion =
```

Comment:
So here we are not checking that `mPendingDataVersion.HasValue()`, unlike in the "cluster changed" case above....  We should be consistent.  Perhaps by having a "UpdateLastConcreteClusterPath", which we call from both there and here (from here with the "invalid path" value) which does this committing logic, updates mLastConcreteClusterPath, etc.

Reply:
updated it, and put it in new function CommitLastWildcardAttributePathVersion